src/main/java/redis/clients/jedis/JedisSentinelPool.java

```diff
@@ -177,14 +180,33 @@ public JedisSentinelPool(String masterName, Set<String> sentinels,
 
   public JedisSentinelPool(String masterName, Set<String> sentinels,
       final GenericObjectPoolConfig<Jedis> poolConfig, final JedisFactory factory) {
+    this(masterName, parseHostAndPorts(sentinels), poolConfig, factory,
+        DefaultJedisClientConfig.builder().build());
+  }
+
+  public JedisSentinelPool(String masterName, Set<HostAndPort> sentinels,
+      final GenericObjectPoolConfig<Jedis> poolConfig, final JedisClientConfig masteClientConfig,
+      final JedisClientConfig sentinelClientConfig) {
+    this(masterName, sentinels, poolConfig, new JedisFactory(masteClientConfig), sentinelClientConfig);
+  }
+
+  public JedisSentinelPool(String masterName, Set<HostAndPort> sentinels,
+      final GenericObjectPoolConfig<Jedis> poolConfig, final JedisFactory factory,
+      final JedisClientConfig sentinelClientConfig) {
     super(poolConfig, factory);
+
     this.poolConfig = poolConfig;
     this.factory = factory;
+    this.sentinelClientConfig = sentinelClientConfig;
 
     HostAndPort master = initSentinels(sentinels, masterName);
     initMaster(master);
   }
 
+  private static Set<HostAndPort> parseHostAndPorts(Set<String> strings) {
```

Comment:
Can we remove `static ` here?

Reply:
Why?

Reply:
We usually do this to provide another static method to use. and of course you can do that in this case. Never mind. : )